griptape/drivers/vector/marqo_vector_store_driver.py

```diff
@@ -0,0 +1,253 @@
+from typing import Optional, List, Dict, Any
+from griptape import utils
+from griptape.drivers import BaseVectorStoreDriver
+from griptape.artifacts import TextArtifact
+import marqo
+from attr import define, field
+import json
+
+
+@define
+class MarqoVectorStoreDriver(BaseVectorStoreDriver):
+    api_key: str = field(kw_only=True)
+    url: str = field(kw_only=True)
+    mq: marqo.Client = field(kw_only=True)
+    index: str = field(kw_only=True)
+
+    def __attrs_post_init__(self):
+        """Initialize the Marqo client with the given API key and URL."""
+        self.mq = marqo.Client(self.url, self.api_key)
+        self.set_index(self.index)
+
+    def set_index(self, index):
+        """Set the index for the Marqo client.
+
+        Args:
+            index (str): The index to set for the Marqo client.
+        """
+        indexes = self.get_indexes()
+
+        if index not in indexes:
+            self.create_index(index)
```

Comment:
Should we log here in case the user expects the index to exist?

Reply:
Good idea, thanks! Just added logging.info()